app/controllers/sites_controller.rb

```diff
@@ -163,7 +136,7 @@ def search_by_collection collection_id, params
     search.after params[:updated_since] if params[:updated_since]
     search.full_text_search params[:search] if params[:search].present?
     search.alerted_search params[:_alert] if params[:_alert].present?
-    search.my_site_search current_user.id unless current_user.can_view_other? collection_id
+    search.my_site_search current_user.id if current_user && !current_user.can_view_other?(collection_id)
```

Comment:
Seem searching my site is contrast with condition, I can search only when I can't view the other otherwise I can't